### DISCUSSION TITLE
Is there a way to get Component through the player?

### DISCUSSION THREAD

**Discussion by Birditch** [KyoriPowered/adventure#922] [General]
In order to popularize translation in my plugin, I used the Gui library (Triumph GUI) that uses the Adventure library when designing the Gui interface. One of the reasons is that I'm going to support the 1.8.8 server, and another reason is that I need to go to Compent for the translation operation.
For example
```
translatable().key("floracore.command.report.abnormal")
```
Then, I found that when I built it directly to Compent and then set it to title.
![image](https://github.com/KyoriPowered/adventure/assets/73741466/6774d717-c39a-4fb4-8339-7e6f6560d157)
I found that it sets the title directly like this.
`Legacy.SERIALIZER.serialize(title)`
Then it returns `floracore.command.report.abnormal` directly in the menu
So, what I'm trying to say is that I want to get the processed Compent directly through the player (Sender).
How should I do that?
Also, I would like to know how I can get Compent to get the processed String through the player (Sender).
If this is not possible, it also means that my Gui must use the source language and cannot be multilingual.
So that's really important to me.
Because I don't seem to see a similar operation in the documentation.
Please help me! Thanks!

**Comment by Birditch** [comment]
![image](https://github.com/KyoriPowered/adventure/assets/73741466/3f0f114b-7f83-45e9-98c4-3212d7625260)

**Comment by kezz** [comment]
Legacy strings do not support translations. You can run the components through the GlobalTranslator yourself if the components are server-side, otherwise you're SOL.

**Reply by Birditch** [reply]
Does that mean it can't be translated for the player?

**Reply by kezz** [reply]
No. You can put translations through the GlobalTranslator to render server-side translations. If it's client-side translations, you should not use legacy formatted strings and use components instead.

**Reply by Birditch** [reply]
Can you give me an example of a GlobalTranslator translation?
If that is not possible, I can accept the result.

**Reply by Birditch** [reply]
But as the title of Chest Gui, it seems that components cannot be used.

**Reply by kezz** [reply]
You can read the javadocs for the GlobalTranslator, I believe it has a render method. If whatever GUI library you are using doesn't support components, that's on them unfortunately.

**Reply by Birditch** [reply]
Okay, thanks.
So the component is actually available in Gui.
Then I should be able to go and learn how IF's components are rendered to it.
The main thing is that IF doesn't support low versions of 1.8.8: (

**Comment by kezz** [comment]
This is also a duplicate of the issue you opened two weeks ago (https://github.com/KyoriPowered/adventure/discussions/908), please do not open duplicate issues asking the same thing.
